packages/lab/src/overlay/Overlay.tsx

```diff
@@ -1,100 +1,158 @@
    */
-  adaExceptions?: ADAExceptions;
-  arrowProps?: ComponentProps<"div">;
-  variant?: "primary" | "secondary";
+  placement?: "bottom" | "top" | "left" | "right";
 }
 
 const withBaseName = makePrefixer("saltOverlay");
 
 export const Overlay = forwardRef<HTMLDivElement, OverlayProps>(
-  (
-    {
-      adaExceptions: { showClose } = {
-        showClose: true,
-      },
-      arrowProps,
+  function Overlay(props, ref) {
+    const {
       children,
       className,
-      open,
-      onOpenChange,
-      variant = "primary",
+      open: openProp,
+      onOpenChange: onOpenChangeProp,
+      content,
+      placement = "top",
       ...rest
-    },
-    ref
-  ) => {
+    } = props;
+
     const targetWindow = useWindow();
     useComponentCssInjection({
       testId: "salt-overlay",
       css: overlayCSS,
       window: targetWindow,
     });
 
-    // Will need to figure out a better way to assign popper id's for the electron windows
-    const id = useId();
-    const Window = usePortalWindow();
+    const [showComponent, setShowComponent] = useState(false);
+
+    const {
+      arrowProps,
+      open,
+      onOpenChange,
+      context,
+      floating,
+      reference,
+      getTriggerProps,
+      getOverlayProps,
+      getOverlayPosition,
+    } = useOverlay({
+      open: openProp,
+      placement,
+      onOpenChange: onOpenChangeProp,
+    });
+
+    const triggerRef = useForkRef(
+      // @ts-ignore
+      isValidElement(children) ? children.ref : null,
+      reference
+    );
+
+    const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+
+    useEffect(() => {
+      if (open && !showComponent) {
+        setShowComponent(true);
+      }
+    }, [open, showComponent]);
 
     const handleCloseButton = () => {
-      onOpenChange?.(false);
+      console.log("close ovl");
+      onOpenChange(false);
     };
 
-    if (!open) {
-      return null;
-    }
-
     return (
-      <Portal>
-        <Window
-          className={clsx(withBaseName(), className, {
-            [withBaseName(variant)]: variant === "secondary",
+      <>
+        {isValidElement(children) &&
+          cloneElement(children, {
+            ...mergeProps(getTriggerProps(), children.props),
+            ref: triggerRef,
           })}
-          id={id}
-          ref={ref}
-          {...rest}
-        >
-          <div
-            className={clsx(withBaseName("content"))}
-            data-testid="overlay-content"
-          >
-            {showClose && (
-              <Button
-                onClick={handleCloseButton}
-                className={withBaseName("close")}
-                variant="secondary"
-              >
-                <CloseIcon
-                  accessible-text="close overlay"
-                  className={withBaseName("closeIcon")}
-                />
-              </Button>
+
+        <FloatingPortal>
```

Comment:
FloatingPortal should be treated as the panel that gets the floating UI props. Is there an issue with the focus manager?

Reply:
updated to use FloatingComponent